tezos/context/src/tezedge_context.rs

```diff
@@ -558,6 +565,14 @@ impl TezedgeIndex {
             },
         }
     }
+
+    fn clone_string_interner_to_repo(&self) -> Result<(), MerkleError> {
```

Comment:
I see this being used in a bunch of places, but I don't fully understand when it needs to run and why it is necessary. Needs a comment.

Reply:
What confuses me the most is that I see it being called in methods that just read from the context. I would imagine that you only need to replace it on commits in cases where the context is modified in a way that introduces new shapes and strings, but that doesn't happen on reads.

Reply:
@tizoc You are right, I only called `clone_string_interner_to_repo` on `commit` and all the tests passed: http://ci.tezedge.com/tezedge/tezedge/2417

Reply:
@sebastiencs ok great. Now that there is no actual cloning one this needs a better name. `update_repository_string_interner` or `syncronize_string_interner_to_repository` or something like that. Also please add a comment explaining what this is doing and why it is needed.

Reply:
@tizoc Done